src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,146 @@
+package duke;
+
+import duke.command.FindCommand;
+import duke.command.Command;
+import duke.command.WhatsupCommand;
+import duke.command.AddCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ByeCommand;
+import duke.command.ListCommand;
+import duke.exception.DukeException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Parser object reads user input strings and informs main Duke object of what command to execute.
+ */
+public class Parser {
+    /**
+     * returns the Command to execute, based on user input.
+     * @param instruction User input.
+     * @return Command to execute.
+     * @throws DukeException If user input is invalid or does not follow specified format.
+     */
+    public static Command parse(String instruction) throws DukeException {
+        if (instruction.equals("")) {
+            throw new DukeException("\t" + "Please use me! Do not leave your commands empty.");
+        }
+
+        String keyword = instruction.split(" ")[0].toLowerCase();
+
+        switch (keyword) {
+        case "bye":
+            return new ByeCommand();
+
+        case "list":
+            return new ListCommand();
+
+        case "done":
+            String[] doneTokens = instruction.split(" ");
+            if (doneTokens.length != 2) {
```

Comment:
the cases in the switch block is not consistently handled. the checking of format of the input can be abstracted elsewhere